ethcore/sync/src/chain/requester.rs

```diff
@@ -129,23 +128,20 @@ impl SyncRequester {
 		trace!(target: "sync", "{} <- GetSnapshotData {:?}", peer_id, chunk);
 		let mut rlp = RlpStream::new_list(1);
 		rlp.append(chunk);
-		SyncRequester::send_request(sync, io, peer_id, PeerAsking::SnapshotData, GET_SNAPSHOT_DATA_PACKET, rlp.out());
+		SyncRequester::send_request(sync, io, peer_id, PeerAsking::SnapshotData, WARP_SYNC_PROTOCOL_ID, GET_SNAPSHOT_DATA_PACKET, rlp.out());
 	}
 
 	/// Generic request sender
-	fn send_request(sync: &mut ChainSync, io: &mut SyncIo, peer_id: PeerId, asking: PeerAsking,  packet_id: PacketId, packet: Bytes) {
+	fn send_request(sync: &mut ChainSync, io: &mut SyncIo, peer_id: PeerId, asking: PeerAsking,  protocol: ProtocolId, packet_id: PacketId, packet: Bytes) {
 		if let Some(ref mut peer) = sync.peers.get_mut(&peer_id) {
 			if peer.asking != PeerAsking::Nothing {
 				warn!(target:"sync", "Asking {:?} while requesting {:?}", peer.asking, asking);
 			}
 			peer.asking = asking;
 			peer.ask_time = Instant::now();
-			// TODO [ToDr] This seems quite fragile. Be careful when protocol is updated.
-			let result = if packet_id >= ETH_PROTOCOL_VERSION_63.1 {
```

Comment:
I meant this if case

Reply:
@tomusdrw ^^

Reply:
That's a different module (and different case), that's why I asked about the original one.
The latter is using `sync.send(` directly which defaults to `network.send(` which in turn defaults to `self.send_protocl(self.protocol)` afair.
So unless I'm missing something both of the methods should get sent using `self.protocol` (which is rpobably ETH_63, not WARP) but somehow get decoded correctly (cause I don't assume they are broken, just some quirk or my misunderstanding causes them to work correctly.)

Reply:
Ok. Made more investigation. As far as I've understood, protocol id for sending packets is set here https://github.com/paritytech/parity-ethereum/blob/06cae8a53556dede67e163325657bee3fba8feda/util/network-devp2p/src/host.rs#L839 in terms of connected callback. I'm not familiar with this code, but it seems, that this PR may break this logic (when protocol id is defined during the session by libp2p).

Reply:
Or may be I'm wrong again :-( For private and consensus packets it's specified explicitly here: https://github.com/paritytech/parity-ethereum/blob/06cae8a53556dede67e163325657bee3fba8feda/ethcore/sync/src/api.rs#L574

Reply:
To be honest, I feel uncomfortable refactoring without enough tests, and from the code alone I think it is not easy to reason about what is happening.

How about I close this PR and begin writing tests for what's already in place? That should give us some peace of mind.